### PR TITLE
Added Algolia Requests for Pages Directly from FE

### LINKED ISSUE #240: Make Algolia requests for Committees page directly from FE
**Is your feature request related to a problem? Please describe.**
At the moment all requests to Algolia go from FE to BE and then to Algolia index engine.

**Describe the solution you'd like**
Implement logic for the Committees page to request data from Algolia right from FE (w/o sending it to BE first)

**Additional context**
https://www.algolia.com/doc/libraries/javascript/v5/

### PR DESCRIPTION
**Pull request by Rajgupta36**
Resolves #240 #237 #238 #239

### Checklist:
- [x] update all pages logic.
- [x] Integrated Algolia JavaScript client for frontend .
- [x] Updated test cases .

----
##media

https://github.com/user-attachments/assets/54472069-5fff-4b00-8f36-ecabdf2641f6

### CODE REVIEW

**Comment by Naveen-Pal** [comment]
Hii @Rajgupta36,
you have also solved the issues #237 , #239 along with #240 
Am i correct?

**Comment by Rajgupta36** [comment]
> Hii @Rajgupta36, you have also solved the issues #237 , #239 along with #240 Am i correct?

yes

File: frontend/src/lib/api.ts [hunk 1]
```diff
@@ -16,3 +22,35 @@ export const loadData = async <T>(
...
+  try {
+    const { results } = await client.search({
+      requests: [
+        {
+          indexName: `${NEST_ENV}_${indexName}`,
+          query,
+          hitsPerPage: 25,
+          page: currentPage - 1,
+          attributesToHighlight: [],
+        },
```
Comment (arkid15r) [inline frontend/src/lib/api.ts]: We want to keep the backend flexibility too:
  - https://github.com/OWASP/Nest/blob/745539cd9af4ee3cd51ebbcbb33da2bd78347b34/backend/apps/owasp/api/search/chapter.py#L13
  - https://github.com/OWASP/Nest/blob/745539cd9af4ee3cd51ebbcbb33da2bd78347b34/backend/apps/owasp/api/search/project.py#L15
  
There should be a way to set params depending on the entity type (project, chapter, issue, etc)

**Review by arkid15r — CHANGES_REQUESTED** [review]
It's important to keep the params flexibility, e.g. for chapters we need to use https://github.com/OWASP/Nest/blob/745539cd9af4ee3cd51ebbcbb33da2bd78347b34/backend/apps/owasp/api/search/chapter.py#L33

File: frontend/src/utils/paramsMapping.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          'idx_related_urls',
+          'idx_summary',
+          'idx_top_contributors',
+          'idx_updated_at',
+          'idx_url',
+        ],
+        typoTolerance: 'min',
+        minProximity: 4,
+      }
+  }
```
Comment (arkid15r) [inline frontend/src/utils/paramsMapping.ts:76]: Do we need `default` here?

**Review by arkid15r — COMMENTED** [review]
Awesome work!

LGTM

File: frontend/src/utils/paramsMapping.ts (hunk 2 above)
Comment (Rajgupta36) [reply frontend/src/utils/paramsMapping.ts:76]: We can implement if we want.

Comment (arkid15r) [reply frontend/src/utils/paramsMapping.ts:76]: it seems to be a good practice, let's follow it

Comment (Rajgupta36) [reply frontend/src/utils/paramsMapping.ts:76]: Okk

**Review by arkid15r — APPROVED** [review]
🚀

**Pull request merged**